src/gp/src/GPMSA.C

```diff
@@ -1034,6 +1048,10 @@ GPMSAFactory<V, M>::setUpEmulator()
       *this->BT_Wy_B_inv,
       *this->KT_K_inv,
       *this->m_opts));
+
+  // FIXME: epic hack.  pls fix.
```

Comment:
This isn't awful.  There's not really a perfect solution to the dichotomy of "put arguments in the constructor so you can enforce consistency and never leave objects in an invalid state" vs "put arguments in setter functions so you don't have to change the constructor every freaking time you add a new option".